### PR TITLE
SLLS-117 Do not use connected mode for notebooks

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java [hunk 1]
```diff
@@ -161,14 +162,23 @@ private void clearIssueCacheAndPublishEmptyDiagnostics(URI f) {
 
   private void analyze(AnalysisTask task, Optional<WorkspaceFolderWrapper> workspaceFolder, Map<URI, VersionedOpenFile> filesToAnalyze) {
     if (workspaceFolder.isPresent()) {
-      // We can only have the same binding for a given folder
+
+      var splitByNotebookOrNot = filesToAnalyze.values().stream()
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java:166]: Maybe `fileGroupsByFileType` or similar is a more descriptive name? `splitByNotebookOrNot` is hard for me to grasp (I speak I 😛)

**Review by sophio-japharidze-sonarsource — APPROVED** [review]
LGTM with minor naming comment

File: src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java [hunk 2]
```diff
@@ -161,14 +162,28 @@ private void clearIssueCacheAndPublishEmptyDiagnostics(URI f) {
...
+
+      var notebooksToAnalyze = new HashMap<URI, VersionedOpenFile>();
+      var nonNotebooksToAnalyze = new HashMap<URI, VersionedOpenFile>();
+      filesToAnalyze.forEach((uri, file) -> {
+        if (openNotebooksCache.isNotebook(uri)) {
+          notebooksToAnalyze.put(uri, file);
+        } else {
+          nonNotebooksToAnalyze.put(uri, file);
+        }
+      });
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java:174]: @sophio-japharidze-sonarsource I believe this is arguably more readable than the functional-style soup I pushed earlier 😅

Comment (sophio-japharidze-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java:174]: YES! 💯

**Review by kirill-knize-sonarsource — APPROVED** [review]

**Pull request merged**